inception-kb/src/test/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseExporterTest.java

```diff
@@ -0,0 +1,155 @@
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+
+import de.tudarmstadt.ukp.clarin.webanno.api.export.ProjectExportRequest;
+import de.tudarmstadt.ukp.clarin.webanno.api.export.ProjectImportRequest;
+import de.tudarmstadt.ukp.clarin.webanno.export.model.ExportedProject;
+import de.tudarmstadt.ukp.clarin.webanno.model.Project;
+import de.tudarmstadt.ukp.inception.kb.exporter.KnowledgeBaseExporter;
+import de.tudarmstadt.ukp.inception.kb.model.KnowledgeBase;
+
+public class KnowledgeBaseExporterTest
+{
+    private static final String TestURLEndpoint = "https://collection.britishmuseum.org/sparql";
+        
+    private @Mock KnowledgeBaseService kbService;
+
+    private Project project;
+    
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    private KnowledgeBaseExporter sut;
+    
+    @Before
+    public void setUp() throws Exception
+    {
+        initMocks(this);
+
+        project = new Project();
+        project.setName("Test Project");
+        when(kbService.getKnowledgeBases(project)).thenReturn(knowledgeBases());
+        
+        ArgumentCaptor<KnowledgeBase> kbCaptor = ArgumentCaptor.forClass(KnowledgeBase.class);
+        when(kbService.getKnowledgeBaseConfig(kbCaptor.capture()))
+                .thenReturn(new SPARQLRepositoryConfig(TestURLEndpoint));
+        
+        sut = new KnowledgeBaseExporter(kbService);
+    }
+    
+    @Test
+    public void thatExportingWorks() throws Exception
+    {
+        // Export the project
+        ProjectExportRequest exportRequest = new ProjectExportRequest();
+        exportRequest.setProject(project);
+        ExportedProject exportedProject = new ExportedProject();
+        sut.exportData(exportRequest, exportedProject, temporaryFolder.getRoot());
+
+        // Import the project again
+        ArgumentCaptor<KnowledgeBase> exportKbCaptor = ArgumentCaptor.forClass(KnowledgeBase.class);
+        ArgumentCaptor<RepositoryImplConfig> cfgCaptor = ArgumentCaptor
+                .forClass(RepositoryImplConfig.class);
+        doNothing().when(kbService).registerKnowledgeBase(exportKbCaptor.capture(),
+                cfgCaptor.capture());
+
+        ProjectImportRequest importRequest = new ProjectImportRequest(true);
+        ZipFile zipFile = mock(ZipFile.class);
+
+        sut.importData(importRequest, project, exportedProject, zipFile);
+
+        // Check how many localKBs have been exported
+        List<KnowledgeBase> exportedKbs = exportKbCaptor.getAllValues();
+        int numOfLocalKBs = exportedKbs.stream()
+                .filter(kb -> kb.getType().equals(RepositoryType.LOCAL))
+                .collect(Collectors.toList()).size();
+
+        // Verify that importData is called as many times as there are localKBs
+        ArgumentCaptor<String> fileNameCaptor = ArgumentCaptor.forClass(String.class);
```

Comment:
You do not need argument captors for that, you can just use matchers like any (https://static.javadoc.io/org.mockito/mockito-core/1.9.5/org/mockito/Matchers.html). You use captors if you want to later assert on the parameters